### PR TITLE
Fix refresh frequency

### PR DESCRIPTION
**Pull request by betodealmeida**
##### SUMMARY

Setting the refresh frequency is not working properly:

1. It doesn't get saved, since it's not sent in the payload to `/superset/save_dash/`;
2. Changing it won't trigger the `hasUnsavedChanges`, so the user might not save the change.

I added the refresh frequency to the payload, and made it so that modifying the refresh frequency sets `hasUnsavedChanges` to true, prompting the user to save the dash.

Note that in order to do the latter I changed the dropdown to only show the option when the user is in edit mode. This can be changed, but I'm worried that users will not know the difference between setting the frequency without entering edit mode (it's not saved, and only applies to the current browser tab) vs. changing it in edit mode (it is saved and applies to all dashboards). Thoughts on this?

##### TEST PLAN

Created new dashboard, changed the refresh frequency. Value is present in the payload, saved in the DB. On next load, the initial value is correctly read from the DB.

##### ADDITIONAL INFORMATION
 

    [ ] Has associated issue:
    [X] Changes UI
    [ ] Requires DB Migration. Confirm DB Migration upgrade and downgrade tested.
    [ ] Introduces new feature or API
    [ ] Removes existing feature or API
    [X] Fixes bug
    [ ] Refactors code
    [ ] Adds test(s)

##### REVIEWERS

@xtinec @mistercrunch

### CODE REVIEW

File: superset/assets/spec/javascripts/dashboard/components/HeaderActionsDropdown_spec.jsx [hunk 1]
```diff
@@ -105,9 +105,9 @@ describe('HeaderActionsDropdown', () => {
       expect(wrapper.find(MenuItem)).toHaveLength(2);
     });
 
-    it('should render the RefreshIntervalModal', () => {
+    it('should not render the RefreshIntervalModal', () => {
```
Comment (williaster) [inline superset/assets/spec/javascripts/dashboard/components/HeaderActionsDropdown_spec.jsx]: should we add tests that assert it renders this in `editMode`?

**Review by williaster — COMMENTED** [review]
This looks good to me except it seems like we should probably have tests that assert the option to update renders in `editMode`?

File: superset/assets/spec/javascripts/dashboard/components/HeaderActionsDropdown_spec.jsx (hunk 1 above)
Comment (betodealmeida) [reply superset/assets/spec/javascripts/dashboard/components/HeaderActionsDropdown_spec.jsx]: We have them already, actually. I didn't have to change them. See https://github.com/apache/incubator-superset/pull/7248/files#diff-c130332fd863f354e0dd6e8e6097f72fR142

**Comment by betodealmeida** [comment]
Thanks, @williaster!

**Comment by graceguo-supercat** [comment]
@betodealmeida this PR makes auto-refresh interval only at edit mode:
Screen Shot 2019-04-10 at 11 49 22 AM

For Superset dashboard, only owner can enable edit mode. check `dash_edit_perm`.
As dashboard viewer (not owner), he should be able to adjust auto-refresh interval at his visit. Just like dashboard filter, any user can apply filter to see different results, but won't affect the original dashboard setting. After this PR, viewer can't have auto-refresh interval anymore.

I think we need to make auto refresh interval choice available for all dashboard viewer.

**Comment by betodealmeida** [comment]
@graceguo-supercat good point. I'll bring back the other option, I just think we need to improve the wording. We should differentiate between setting a refresh on the dashboard the user is currently seeing, versus saving it with the dashboard metadata.

**Pull request merged**